aspnetcore/mvc/razor-pages/filter/sample/PageFilter/Filters/AddHeaderAttribute.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace PageFilter.Filters
+{
+    public class AddHeaderAttribute  : ResultFilterAttribute
+    {
+        private readonly string _name;
+        private readonly string _value;
+
+        public AddHeaderAttribute (string name, string value)
+        {
+            _name = name;
+            _value = value;
+        }
+
+        public override async Task OnResultExecutionAsync(
```

Comment:
Could you change this to `public override void `OnResultExecuting(ResultExecutingContext context)`? Would be simpler than the async overload.